### PR TITLE
count xform submissions and send to datadog

### PR DESCRIPTION
**Pull request by snopoke**
@benrudolph @czue @dannyroberts 

cc @sravfeyn

### CODE REVIEW

**Comment by czue** [comment]
looks good. are we quite sure that there isn't a major performance hit to adding this?

**Comment by snopoke** [comment]
Yes, performance hit is tiny. You can read more here: https://www.datadoghq.com/blog/statsd/

**Comment by snopoke** [comment]
@benrudolph @esoergel @NoahCarnahan @dannyroberts we're GTG to start creating custom metrics. More docs here: http://docs.datadoghq.com/guides/metrics/

**Comment by benrudolph** [comment]
awesome, mergin

File: corehq/util/datadog/__init__.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        return self.ret
+
+
+class MockStatsd(object):
+    def __getattribute__(self, item):
+        if item == 'timed':
+            def no_op_decorator(fn):
+                return fn
+            return Callable(item, ret=no_op_decorator)
+        return Callable(item)
```
Comment (esoergel) [inline corehq/util/datadog/__init__.py]: What made you choose to use `__getattribute__` over implementing `timed` and `increment` methods?  Is this just going to be called with a bunch of weird parameters?

File: corehq/util/datadog/__init__.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class Callable(object):
+    def __init__(self, name, ret=None):
+        self.name = name
+        self.ret = ret
+
+    def __call__(self, *args, **kwargs):
+        logger.debug("mock call: statsd.%s(*%s, **%s)", self.name, args, kwargs)
+        return self.ret
```
Comment (esoergel) [inline corehq/util/datadog/__init__.py]: I like this class - I've wanted something like it a couple times.

File: corehq/util/datadog/__init__.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+from django.conf import settings
+import logging
+
+logger = logging.getLogger(__name__)
+
+try:
+    from datadog.dogstatsd.base import DogStatsd
+    use_real_statsd = True
+except ImportError:
+    use_real_statsd = False
```
Comment (esoergel) [inline corehq/util/datadog/__init__.py]: What environments will not have `datadog` available?  Locally?

**Comment by esoergel** [comment]
This is awesome!  How much restraint should we show in using it, any?  Say for instance I wanted to see how often a function was being called in production, could I use a similar approach for that?

File: corehq/util/datadog/__init__.py (hunk 1 above)
Comment (snopoke) [inline corehq/util/datadog/__init__.py]: There are other functions that can also be called and I didn't want to have to implement them all (lazy)

File: corehq/util/datadog/__init__.py (hunk 3 above)
Comment (snopoke) [inline corehq/util/datadog/__init__.py]: Yea, this should only every happen locally since the datadog dependency is in prod-requirements.

**Comment by snopoke** [comment]
I don't think we should use it for one off / short term tracking. What you're talking about could be accomplished with a variation of `soft_assert`. This is more for getting metrics that will help us diagnose issue etc.

**Comment by snopoke** [comment]
Seems like this is officially called the 'Bridge' pattern: https://sourcemaking.com/design_patterns/bridge

**Pull request merged**